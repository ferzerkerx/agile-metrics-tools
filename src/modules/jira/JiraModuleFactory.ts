import { CollectorModuleFactory, Utils } from '../../metrics';
import { JiraCollectorConfig, JiraMetricItem } from './collector/Types';
import { JiraService } from './Types';
import { JiraClientImpl } from './JiraClientImpl';
import { JiraServiceImpl } from './JiraServiceImpl';
import { JiraCollectorsService } from './collector/JiraCollectorsService';
import { JiraRepositoryImpl } from './JiraRepositoryImpl';

export class JiraModuleFactory
  implements CollectorModuleFactory<JiraCollectorConfig, JiraMetricItem> {
  private static jiraService(): JiraService {
    Utils.checkEnvVar('JIRA_HOST', 'JIRA_API_TOKEN');
    const jiraClient = new JiraClientImpl({
      host: `${process.env.JIRA_HOST}`,
      apiToken: `${process.env.JIRA_API_TOKEN}`
    });
    const jiraRepository = new JiraRepositoryImpl(jiraClient);
    return new JiraServiceImpl(jiraRepository);
  }

  collectorInstance(): JiraCollectorsService {
    return new JiraCollectorsService(JiraModuleFactory.jiraService());
  }

  collectorConfiguration(obj: any): JiraCollectorConfig {
    return new JiraCollectorConfig(obj);
  }
}
