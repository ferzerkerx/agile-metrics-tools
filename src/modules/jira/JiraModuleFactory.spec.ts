import { JiraModuleFactory } from "./JiraModuleFactory";

describe("JiraModuleFactory", () => {
  const jiraModuleFactory: JiraModuleFactory = new JiraModuleFactory();

  beforeEach(() => {
    jest.resetModules();
    delete process.env.JIRA_HOST;
    delete process.env.JIRA_API_TOKEN;
  });

  it("should create collector", () => {
    process.env.JIRA_HOST = "SOME_TOKEN";
    process.env.JIRA_API_TOKEN = "JIRA_API_TOKEN";
    const collectorInstance = jiraModuleFactory.collectorInstance();
    expect(collectorInstance).not.toBeNull();
  });

  it("should fail to create collector", () => {
    expect(jiraModuleFactory.collectorInstance).toThrowErrorMatchingSnapshot();
  });

  it("should create collectorConfiguration", () => {
    const collectorConfiguration = jiraModuleFactory.collectorConfiguration({});
    expect(collectorConfiguration).not.toBeNull();
  });
});
