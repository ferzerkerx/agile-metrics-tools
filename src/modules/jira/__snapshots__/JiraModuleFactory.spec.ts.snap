// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JiraModuleFactory should fail to create collector 1`] = `"env.JIRA_HOST not set!"`;
